contrib/platform/src/com/sun/jna/platform/win32/WbemcliUtil.java

```diff
@@ -0,0 +1,783 @@
+     * 6 and 7 are where data is queried and received.
+     */
+
+    /**
+     * Initializes COM library and sets security to impersonate the local user
+     */
+    public static void initCOM() {
+        HRESULT hres = null;
+        // Step 1: --------------------------------------------------
+        // Initialize COM. ------------------------------------------
+        if (!isComInitialized()) {
+            hres = Ole32.INSTANCE.CoInitializeEx(null, Ole32.COINIT_MULTITHREADED);
+            switch (hres.intValue()) {
+            // Successful local initialization
+            case COMUtils.S_OK:
+                comInitialized = true;
+                break;
+            // COM was already initialized
+            case COMUtils.S_FALSE:
+            case WinError.RPC_E_CHANGED_MODE:
+                break;
+            // Any other results is an error
+            default:
+                throw new Wbemcli.WbemcliException("Failed to initialize COM library.", hres.intValue());
+            }
+        }
+        // Step 2: --------------------------------------------------
+        // Set general COM security levels --------------------------
+        if (!isSecurityInitialized()) {
+            hres = Ole32.INSTANCE.CoInitializeSecurity(null, new NativeLong(-1), null, null,
+                    Ole32.RPC_C_AUTHN_LEVEL_DEFAULT, Ole32.RPC_C_IMP_LEVEL_IMPERSONATE, null, Ole32.EOAC_NONE, null);
+            // If security already initialized we get RPC_E_TOO_LATE
+            // This can be safely ignored
+            if (COMUtils.FAILED(hres) && hres.intValue() != WinError.RPC_E_TOO_LATE) {
+                Ole32.INSTANCE.CoUninitialize();
+                throw new Wbemcli.WbemcliException("Failed to initialize security.", hres.intValue());
+            }
+            securityInitialized = true;
+        }
+    }
+
+    /**
+     * Obtains a locator to the WMI server and connects to the specified
+     * namespace
+     *
+     * @param namespace
+     *            The namespace to connect to
+     * @return A service representing the connected namespace, which can be
+     *         queried. This service may be re-used for multiple queries and
+     *         should be released by the user
+     */
+    public static IWbemServices connectServer(String namespace) {
+        PointerByReference pSvc = new PointerByReference();
+        // Step 3: ---------------------------------------------------
+        // Obtain the initial locator to WMI -------------------------
+        IWbemLocator loc = IWbemLocator.create();
+
+        // Step 4: -----------------------------------------------------
+        // Connect to WMI through the IWbemLocator::ConnectServer method
+        // Connect to the namespace with the current user and obtain pointer
+        // pSvc to make IWbemServices calls.
+        HRESULT hres = loc.ConnectServer(new BSTR(namespace), null, null, null, null, null, null, pSvc);
```

Comment:
Don't use the `BSTR` constructor. `BSTR` need to be allocated via `OleAuto.INSTANCE.SysAllocString` and they need to be freed with `OleAuto.INSTANCE.SysFreeString`.